Assets/Scripts/Button Cell.cs

```diff
@@ -0,0 +1,50 @@
+
+using UnityEngine;
+using TMPro;
+using UnityEngine.UI;
+
+public class ButtonCell : Cell
+{
+    [SerializeField] private TextMeshProUGUI cellText;
+    [SerializeField] private Image image;
+    private System.Action callback;
+    private ButtonCellData bData = new();
+
+    public override void Configure(CellData data)
+    {
+        bData = data as ButtonCellData;
+        callback = bData.action;
+        SetTextValue(bData.text);
+        if (bData.isImageActive)
+        {
+            image.color = Color.gray;
+        }
+        else
+        {
+            image.color = Color.white;
+        }
+    }
+    public void OnClick()
+    {
+        callback?.Invoke();
```

Comment:
I like this use of null check on callback

Reply:
Resolved